lib/modules/forgot_password/forgot_password_view.dart

```diff
@@ -0,0 +1,208 @@
+  static const showingResetSuccessfullyNotificationDuration =
+      Duration(milliseconds: 600);
+  static const dismissResetSuccessfullyNotificationDelayDuration =
+      Duration(seconds: 2);
+
+  static const emailTextFieldKey = Key("email_text_field_key");
+  static const resetButtonKey = Key("reset_button_key");
+  static const resetSuccessfullyNotificationSlideTransitionKey =
+      Key("reset_successfully_notification_slide_transition");
+  static const resetSuccessfullyNotificationKey =
+      Key("reset_successfully_notification");
+
+  static const resetSuccessfullyNotificationSlideTransitionBeginOffset =
+      Offset(0, -1);
+  static const resetSuccessfullyNotificationSlideTransitionEndOffset =
+      Offset.zero;
+
+  void clearEmailTextField();
+
+  void setResetButton({required bool isEnabled});
+
+  void showResetSuccessfullyNotification();
+}
+
+abstract class ForgotPasswordViewDelegate implements AlertViewMixinDelegate {
+  BehaviorSubject<String> get emailTextFieldDidChange;
+
+  BehaviorSubject<String> get resetButtonDidTap;
+}
+
+class ForgotPasswordViewImpl extends StatefulWidget {
+  const ForgotPasswordViewImpl({Key? key}) : super(key: key);
+
+  @override
+  _ForgotPasswordViewImplState createState() => _ForgotPasswordViewImplState();
+}
+
+class _ForgotPasswordViewImplState extends ViewState<ForgotPasswordViewImpl,
+        ForgotPasswordModule, ForgotPasswordViewDelegate>
+    with
+        ProgressHUDViewMixin<ForgotPasswordViewDelegate>,
+        SingleTickerProviderStateMixin,
+        AlertViewMixin
+    implements ForgotPasswordView {
+  late final AnimationController _notificationAnimationController =
+      AnimationController(
+    duration: ForgotPasswordView.showingResetSuccessfullyNotificationDuration,
+    vsync: this,
+  );
+
+  late final Animation<Offset> _notificationAnimation = Tween<Offset>(
+    begin: ForgotPasswordView
+        .resetSuccessfullyNotificationSlideTransitionBeginOffset,
+    end: ForgotPasswordView
+        .resetSuccessfullyNotificationSlideTransitionEndOffset,
+  ).animate(_notificationAnimationController);
+
+  final _emailTextController = TextEditingController();
+  final _isResetButtonEnabled = BehaviorSubject<bool>.seeded(false);
+
+  CancelableOperation? _dismissNotificationOperation;
+
+  @override
+  void initState() {
+    _emailTextController.addListener(() {
+      delegate?.emailTextFieldDidChange.add(_emailTextController.text);
+    });
+    _notificationAnimationController.addStatusListener((status) {
+      if (status != AnimationStatus.completed) return;
+
+      // Set a timeout to slide out the notification
+      _dismissNotificationOperation?.cancel();
+      _dismissNotificationOperation = CancelableOperation.fromFuture(
+          Future.delayed(
+              ForgotPasswordView
+                  .dismissResetSuccessfullyNotificationDelayDuration, () {
+        _dismissResetSuccessfullyNotification();
+      }));
+    });
+    super.initState();
```

Comment:
like @manh-t mentioned, there are too many logic/integrate code in this [UI] PR

Reply:
@sleepylee sorry, I till learn to separate UI and Integrate into difference PR